contracts/SortitionPool.sol

```diff
@@ -56,13 +83,29 @@ contract SortitionPool is SortitionTree {
 
     require(eligibleWeight != inPoolWeight, "Operator already up to date");
 
+    Rewards.updateOperatorRewards(operator, uint32(inPoolWeight));
+
     if (eligibleWeight == 0) {
       removeOperator(operator);
     } else {
       updateOperator(operator, eligibleWeight);
     }
   }
 
+  function setRewardIneligibility(address[] calldata operators, uint256 until)
+    public
+  {
+    require(
+      msg.sender == poolParams.owner,
+      "Only owner may set operators ineligible"
+    );
+    Rewards.setIneligible(operators, until);
+  }
```

Comment:
What is the gas cost of calling this function? We do not have `address[]` but `uint32[]` identifiers because it's how we store group members info on-chain. See the signature here:
https://github.com/keep-network/sortition-pools/blob/6f98f873508c3742663da4c086048451ea3e578d/contracts/SortitionPool.sol#L115

Sure, we can turn ids into operator addresses but the cost could be significant and we sometimes need to set reward ineligibility in relay entry submit transaction. In the worst case for 63 groups members.

Is there any gas station pattern we could use here?

Reply:
If the gas cost is high and we can do nothing about it, is there a way we can reward the caller of this function (assuming it is a sortition pool member) by moving some `available` rewards from ineligible operators to the notifier? This way, the ineligible operator loses twice: future rewards for some time and gets some cut on not yet withdrawn rewards.

Reply:
The gas cost is roughly 10k per operator that is made ineligible; e.g. 25 operators costs ~250k gas. I can update it to use IDs just as easily instead of addresses, that was an oversight.

Reply:
Is there a way to return some kind of information allowing us to estimate "impact" of calling `setIneligible`?
For example, calling `setIneligible` for operator who was eligible for rewards and making in ineligible for 2 weeks has a higher "impact" than calling `setIneligible` for operator that was marked 10 minutes ago as ineligible for rewards for 2 weeks.

Say we have one group with 25 inactive members and relay entry is requested every 5 minutes. If we need to pay ~250k gas to constantly prolong their ineligibility, we end up with a situation where the cost of prolonging their ineligibility is higher than what we save on marking them as ineligible. 

In https://github.com/keep-network/keep-core/issues/2725 we started working on a solution when "mark inactive as ineligible for rewards" happens outside of DKG / relay entry transaction but it feels that we really need a way to measure the impact to decide what should be the reward for calling this function. If the impact is low, the reward is low, and we can skip marking one as ineligible (e.g. when it already happened 10 minutes ago).

Reply:
Resolving as no longer relevant, we decided to use the same mechanism as for tBTC v2 heartbeats to lower costs, see https://github.com/keep-network/keep-core/issues/2725.